### PR TITLE
jax versions of all operators with jax_finufft

### PR DESCRIPTION
**Pull request by TomHilder**
Added `jax_operators.py` with `jax` + `jax_finufft` versions of all (1D, 2D, 3D) operators. 

Some considerations for the `jax` versions of the operators:
1. Using single precision is pretty bad in terms of accuracy. This is unfortunate because single precision is the default for `jax`. The precision is [easily changed to double](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision), although not all `jax` features are supported for double precision for all hardware. 
2. The 1D operator will not support GPU as the 1D transform in `jax_finufft` does not (see flatironinstitute/jax-finufft#125).

No tests written as of yet as I'm unfamiliar with `uv`, and I'm not sure how to include more dependencies for the tests.

### CODE REVIEW

**Comment by TomHilder** [comment]
Tests are failing since the test environment doesn't have `jax` or `jax_finufft`. Although no tests rely on that so idk why it matters? Bit rusty on unit testing, probably will need some input.
